src/Service/ComponentenCatalogusService.php

```diff
@@ -21,343 +21,102 @@ class ComponentenCatalogusService
     ) {
-        $this->entityManager   = $entityManager;
-        $this->callService     = $callService;
-        $this->cacheService    = $cacheService;
-        $this->syncService     = $syncService;
-        $this->mappingService  = $mappingService;
-        $this->donService      = $donService;
-        $this->pluginLogger    = $pluginLogger;
-        $this->resourceService = $resourceService;
+        $this->pluginLogger        = $pluginLogger;
+        $this->resourceService     = $resourceService;
+        $this->getResourcesService = $getResourcesService;
+        $this->data                = [];
+        $this->configuration       = [];
 
     }//end __construct()
 
 
     /**
-     * Get applications through the products of https://componentencatalogus.commonground.nl/api/products.
+     * Get all applications or one application through the products of https://componentencatalogus.commonground.nl/api/products.
      *
-     * @return array|null
-     */
-    public function getApplications(): ?array
-    {
-        $result = [];
-        // Do we have a source?
-        $source = $this->resourceService->getSource('https://opencatalogi.nl/source/oc.componentencatalogus.source.json', 'open-catalogi/open-catalogi-bundle');
-
-        $applications = $this->callService->getAllResults($source, '/products');
-
-        $this->pluginLogger->info('Found '.count($applications).' applications');
-        foreach ($applications as $application) {
-            $result[] = $this->importApplication($application);
-        }
-
-        $this->entityManager->flush();
-
-        return $result;
-
-    }//end getApplications()
-
-
-    /**
-     * Get an application through the products of https://componentencatalogus.commonground.nl/api/products/{id}.
-     *
-     * @param string $applicationId The id of the application to look for.
+     * @param array|null  $data          The data array from the request
+     * @param array|null  $configuration The configuration array from the request
+     * @param string|null $applicationId The given application id
      *
      * @return array|null
      */
-    public function getApplication(string $applicationId): ?array
+    public function getComponentenCatalogusApplications(?array $data=[], ?array $configuration=[], ?string $applicationId=null): ?array
     {
-        // Do we have a source?
-        $source = $this->resourceService->getSource('https://opencatalogi.nl/source/oc.componentencatalogus.source.json', 'open-catalogi/open-catalogi-bundle');
+        $this->data          = $data;
+        $this->configuration = $configuration;
 
-        $this->pluginLogger->info('Getting application '.$applicationId);
-        $response = $this->callService->call($source, '/products/'.$applicationId);
+        // Get the source from the configuration array.
+        $source   = $this->resourceService->getSource($this->configuration['source'], 'open-catalogi/open-catalogi-bundle');
```

Comment:
Just a reminder that the resourceService functions can return null and $source = null (or any other case like this ending up in a null result) can break your code.